datafusion/expr/src/expr.rs

```diff
@@ -434,7 +450,25 @@ impl fmt::Debug for Expr {
             Expr::Negative(expr) => write!(f, "(- {:?})", expr),
             Expr::IsNull(expr) => write!(f, "{:?} IS NULL", expr),
             Expr::IsNotNull(expr) => write!(f, "{:?} IS NOT NULL", expr),
-            Expr::Exists(subquery) => write!(f, "EXISTS ({:?})", subquery),
+            Expr::Exists { subquery, negated } => {
+                if *negated {
+                    write!(f, "NOT EXISTS ({:?})", subquery)
+                } else {
+                    write!(f, "EXISTS ({:?})", subquery)
+                }
+            }
```

Comment:
I am not sure if you think this is cleaner, but you could potentially use multiple pattern matches like this (untested)

```suggestion
            Expr::Exists { subquery, negated: true } => write!(f, "NOT EXISTS ({:?})", subquery),
            Expr::Exists { subquery, negated: false } => write!(f, "EXISTS ({:?})", subquery),
```

The same comment applies for `Expr::InSubquery`